src/number_theory/arithmetic_function.lean

```diff
@@ -289,9 +314,19 @@ begin
   by_cases h1 : y.fst = 0,
   { simp [function.comp_apply, h1] },
   { simp only [h1, mul_one, one_mul, prod.fst_swap, function.embedding.coe_fn_mk, prod.snd_swap,
-      if_false, zeta_apply, zero_hom.coe_mk] }
+      if_false, zeta_apply, zero_hom.coe_mk, nat_coe_apply, cast_one] }
 end
 
+@[simp]
```

Comment:
```suggestion
```
I don't think this should be in the default simp-set. The RHS isn't really that simple.